CMakeLists.txt

```diff
@@ -219,34 +221,39 @@ endif()
-add_test(NAME cargo_doc COMMAND cargo doc --release --all-features --target-dir ${CARGO_TARGET_DIR})
-add_test(NAME cargo_clippy COMMAND cargo clippy --release --all-features --target-dir ${CARGO_TARGET_DIR} -- -D warnings)
+if(BUILD_TESTS)
+    # Add CMake tests for `cargo test/clippy/fmt/doc`.
+    add_test(NAME cargo_tests COMMAND cargo test --release --all-features --target-dir ${CARGO_TARGET_DIR})
+    add_test(NAME cargo_doc COMMAND cargo doc --release --all-features --target-dir ${CARGO_TARGET_DIR})
+    add_test(NAME cargo_clippy COMMAND cargo clippy --release --all-features --target-dir ${CARGO_TARGET_DIR} -- -D warnings)
+endif()
 
 if(CMAKE_RUSTC_WRAPPER)
     list(APPEND CARGO_ENV "RUSTC_WRAPPER=set:${CMAKE_RUSTC_WRAPPER}")
 endif()
 
-set_tests_properties(cargo_tests cargo_clippy PROPERTIES
-    ENVIRONMENT_MODIFICATION "${CARGO_ENV}"
-)
-set_tests_properties(cargo_doc PROPERTIES
-    ENVIRONMENT_MODIFICATION "${CARGO_ENV};RUSTDOCFLAGS=set:--deny=warnings"
-)
-
-# Ensure test inputs and outputs are formatted
-file(GLOB CXX_QT_GEN_TEST_INPUTS ${CMAKE_CURRENT_SOURCE_DIR}/crates/cxx-qt-gen/test_inputs/*.rs)
-file(GLOB CXX_QT_GEN_TEST_OUTPUTS ${CMAKE_CURRENT_SOURCE_DIR}/crates/cxx-qt-gen/test_outputs/*.rs)
-add_test(NAME cxx_qt_gen_test_inputs_gen COMMAND rustfmt --check ${CXX_QT_GEN_TEST_INPUTS})
-add_test(NAME cxx_qt_gen_test_outputs_gen COMMAND rustfmt --check ${CXX_QT_GEN_TEST_OUTPUTS})
-
-# Add test which checks that a build rerun doesn't recompile and uses caches instead
-add_test(NAME cargo_build_rerun COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/check_cargo_build_rerun.sh" "${CMAKE_CURRENT_SOURCE_DIR}")
-
-# Ensure that cargo_build_rerun doesn't run while we are already building
-set_tests_properties(cargo_build_rerun PROPERTIES RUN_SERIAL TRUE)
-
 add_subdirectory(book)
 add_subdirectory(examples)
-add_subdirectory(tests)
+
+if(BUILD_TESTS)
+    set_tests_properties(cargo_tests cargo_clippy PROPERTIES
+        ENVIRONMENT_MODIFICATION "${CARGO_ENV}"
+    )
+    set_tests_properties(cargo_doc PROPERTIES
+        ENVIRONMENT_MODIFICATION "${CARGO_ENV};RUSTDOCFLAGS=set:--deny=warnings"
+    )
+    
+    # Ensure test inputs and outputs are formatted
+    file(GLOB CXX_QT_GEN_TEST_INPUTS ${CMAKE_CURRENT_SOURCE_DIR}/crates/cxx-qt-gen/test_inputs/*.rs)
+    file(GLOB CXX_QT_GEN_TEST_OUTPUTS ${CMAKE_CURRENT_SOURCE_DIR}/crates/cxx-qt-gen/test_outputs/*.rs)
+    add_test(NAME cxx_qt_gen_test_inputs_gen COMMAND rustfmt --check ${CXX_QT_GEN_TEST_INPUTS})
+    add_test(NAME cxx_qt_gen_test_outputs_gen COMMAND rustfmt --check ${CXX_QT_GEN_TEST_OUTPUTS})
+    
+    # Add test which checks that a build rerun doesn't recompile and uses caches instead
+    add_test(NAME cargo_build_rerun COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/scripts/check_cargo_build_rerun.sh" "${CMAKE_CURRENT_SOURCE_DIR}")
+    
+    # Ensure that cargo_build_rerun doesn't run while we are already building
+    set_tests_properties(cargo_build_rerun PROPERTIES RUN_SERIAL TRUE)
```

Comment:
can we add these to the preview if block as they are linked to the ` add_test(NAME cargo_clippy` etc some are modifying the env for those and the rest are adding more cargo related tests

Then just leave the add_subdir of tests for this block

Reply:
i meant the preview `if(BUILD_TESTING)` block :-)  so around line 228

Reply:
oh my